controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1218,6 +1219,9 @@ func (r *BareMetalHostReconciler) manageHostPower(prov provisioner.Provisioner,
 	}
 
 	if provResult.ErrorMessage != "" {
+		if !desiredPowerOnState && desiredRebootMode == metal3v1alpha1.RebootModeSoft {
```

Comment:
We don't attempt a soft power off if `info.host.Status.ErrorType == metal3v1alpha1.PowerManagementError`:
https://github.com/metal3-io/baremetal-operator/blob/main/pkg/provisioner/ironic/ironic.go#L1803
so we should also check that here.